src/main/java/io/codemc/bot/commands/CmdRemove.java

```diff
@@ -0,0 +1,122 @@
+import java.util.concurrent.CompletableFuture;
+
+public class CmdRemove extends BotCommand{
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(CmdRemove.class);
+
+    public CmdRemove(CodeMCBot bot){
+        super(bot);
+
+        this.name = "remove";
+        this.help = "Revoke a user's Author status.";
+
+        this.allowedRoles = bot.getConfigHandler().getLongList("allowed_roles", "commands", "service");
+
+        this.options = List.of(
+                new OptionData(OptionType.USER, "user", "The user that is having their status revoked.").setRequired(true),
+                new OptionData(OptionType.STRING, "username", "The Jenkins/Nexus username of the user. If blank, defaults to their server nickname.").setRequired(false)
+        );
+    }
+
+    @Override
+    public void withModalReply(SlashCommandEvent event){}
+
+    @Override
+    public void withHookReply(InteractionHook hook, SlashCommandEvent event, Guild guild, Member member){
+        Member user = event.getOption("user", OptionMapping::getAsMember);
+
+        if (user == null) {
+            CommandUtil.EmbedReply.from(hook).error("Invalid User was provided!").send();
+            return;
+        }
+
+        String username = event.getOption("username", user.getNickname(), OptionMapping::getAsString);
+        Role authorRole = guild.getRoleById(bot.getConfigHandler().getLong("author_role"));
+
+        if (authorRole == null) {
+            CommandUtil.EmbedReply.from(hook).error("The Author role is not set up correctly!").send();
+            return;
+        }
+
+        boolean hasAuthor = user.getRoles().stream()
+                .anyMatch(role -> role.getIdLong() == authorRole.getIdLong());
+
+        if (!hasAuthor) {
+            CommandUtil.EmbedReply.from(hook).error("The user is not an Author!").send();
+            return;
+        }
+
+        if (JenkinsAPI.getJenkinsUser(username).isBlank()) {
+            CommandUtil.EmbedReply.from(hook).error("The user does not have a Jenkins account!").send();
+            return;
+        }
+
+        JenkinsAPI.deleteUser(username);
+
+        CompletableFuture<Boolean> deleted = new CompletableFuture<>();
+        deleted.handleAsync((success, ex) -> {
+            if (ex != null) {
+                CommandUtil.EmbedReply.from(hook).error("Failed to delete Nexus User!").send();
+                LOGGER.error("Failed to delete Nexus User '{}'!", username, ex);
+                return false;
+            }
+
+            return true;
+        });
+        NexusAPI.deleteNexus(username, JavaContinuation.create(deleted));
+
+        guild.removeRoleFromMember(user, authorRole)
+                .reason("[Access Request] Author Status revoked by " + member.getUser().getAsTag())
+                .queue(
+                        v -> CommandUtil.EmbedReply.from(hook)
+                                .success("Revoked Author Status from " + member.getUser().getEffectiveName() + "!")

```

Comment:
Wrong member used here.
```suggestion
                .reason("[Access Request] Author Status revoked by " + user.getUser().getAsTag())
                .queue(
                        v -> CommandUtil.EmbedReply.from(hook)
                                .success("Revoked Author Status from " + user.getUser().getEffectiveName() + "!")

```